### PR TITLE
Add custom network deletion + tests

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

File: src/features/Settings/components/AddOrEditNetworkNode.tsx [hunk 1]
```diff
@@ -200,15 +212,18 @@ export default function AddOrEditNetworkNode({
...
+      deleteNetwork(networkId);
+    } else {
       deleteNode(editNode!.name, networkId);
-      onComplete();
-    },
-    [deleteNode, onComplete]
-  );
+    }
+    onComplete();
+  };
```
Comment (vilsbole) [inline src/features/Settings/components/AddOrEditNetworkNode.tsx:226]: Now that we have redux, let's dispatch an action with network id and let the reducer handle the rest.

File: src/services/Store/Network/useNetworks.tsx [hunk 2]
```diff
@@ -82,38 +93,8 @@ function useNetworks() {
     updateNetwork(networkUpdate);
     EthersJS.updateEthersInstance(networkUpdate);
   };
-  const deleteNode = (nodeName: string, network: Network | NetworkId) => {
```
Comment (vilsbole) [inline src/services/Store/Network/useNetworks.tsx:85]: 👍

**Review by vilsbole — CHANGES_REQUESTED** [review]

File: src/features/Settings/components/AddOrEditNetworkNode.test.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import { fAccounts, fAssets, fContacts, fNetwork, fNetworks, fSettings } from '@fixtures';
+import { DataContext, NetworkUtils, ProviderHandler, StoreProvider } from '@services';
+import { translateRaw } from '@translations';
+import { NodeType } from '@types';
+import { noOp } from '@utils';
+
+import AddOrEditNetworkNode from './AddOrEditNetworkNode';
+
+// Mock getCurrentBlock
+ProviderHandler.prototype.getCurrentBlock = jest.fn().mockResolvedValue(11386255);
```
Comment (vilsbole) [inline src/features/Settings/components/AddOrEditNetworkNode.test.tsx:22]: I believe this mocks the method for the entire test suite.

File: src/services/Store/store/network.slice.ts [hunk 4]
```diff
@@ -37,19 +39,52 @@ const slice = createSlice({
...
+      const newNodes = [...nodes.filter((n) => n.name !== nodeName)];
+
+      const newSelectedNode = (() => {
+        if (
+          network.selectedNode === nodeName &&
+          (network.selectedNode === network.autoNode || network.autoNode === undefined)
+        ) {
+          return newNodes[0]?.name;
+        } else if (network.selectedNode === nodeName) {
+          return network.autoNode;
```
Comment (vilsbole) [inline src/services/Store/store/network.slice.ts:59]: Please add test case for this branch

File: src/services/Store/store/network.slice.ts [hunk 5]
```diff
@@ -37,19 +39,52 @@ const slice = createSlice({
...
+        }
+        return network.selectedNode;
```
Comment (vilsbole) [inline src/services/Store/store/network.slice.ts:61]: Please add test case for this branch

File: src/services/Store/store/network.slice.ts [hunk 6]
```diff
@@ -60,3 +95,51 @@ export default slice;
 
 export const getNetworks = createSelector([getAppState], (s) => s[slice.name]);
 export const getDefaultNetwork = createSelector(getNetworks, find(propEq('id', DEFAULT_NETWORK)));
+export const getNetwork = (network: NetworkId) =>
+  createSelector(getNetworks, find(propEq('id', network)));
+export const canDeleteNode = (networkId: NetworkId) =>
+  createSelector([getAppState], (state) => {
+    const network = state.networks.find((n) => n.id === networkId)!;
+
+    if (network.isCustom && network.nodes.length === 1) {
```
Comment (vilsbole) [inline src/services/Store/store/network.slice.ts:104]: Include test for selector.

**Review by vilsbole — CHANGES_REQUESTED** [review]

File: src/features/Settings/components/AddOrEditNetworkNode.test.tsx (hunk 3 above)
Comment (FrederikBolding) [reply src/features/Settings/components/AddOrEditNetworkNode.test.tsx:22]: Yes, which is intended.

**Review by vilsbole — APPROVED** [review]

**Pull request merged**